nodejs-week1/jianxin

jianxinz233 opened the pull request:
meal sharing app routes setting

jianxinz233 committed:
nodejs-w1-hw

jianxinz233 committed:
fix the npm install issue

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+apiRouter.get("/future-meals", async (req, res) => {
+  try{
+    const GET_FUTURE_MEALS_QUERY = "SELECT * FROM meal WHERE `when` > NOW();";
+    const [futureMeals] = await knex.raw(GET_FUTURE_MEALS_QUERY);
```
```suggestion
    const [futureMeals, schema] = await knex.raw(GET_FUTURE_MEALS_QUERY);
```
The `knex.raw()` call returns the rows as an array as the first return value, and then the schema as the second one. You should only return the result, not the schema.

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+apiRouter.get("/past-meals", async (req, res) => {
+  try{
+    const GET_PAST_MEALS_QUERY = "SELECT * FROM meal WHERE `when` < NOW();";
+    const [pastMeals] = await knex.raw(GET_PAST_MEALS_QUERY);
```
Same applies here.

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+apiRouter.get("/all-meals", async (req, res) => {
+  try{
+    const GET_ALL_MEALS_QUERY = "SELECT * FROM meal ORDER BY ID;";
+    const [allMeals] = await knex.raw(GET_ALL_MEALS_QUERY);
```
And here.

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+apiRouter.get("/first-meal", async (req, res) => {
+  try{
+    const GET_FIRST_MEAL_QUERY = "SELECT * FROM meal ORDER BY ID LIMIT 1;";
+    const [firstMeal] = await knex.raw(GET_FIRST_MEAL_QUERY);
```
```suggestion
    const [firstMeal, schema] = await knex.raw(GET_FIRST_MEAL_QUERY);
```

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+    if(!firstMeal || firstMeal.length === 0) {
+      return res.status(404).json({ error: "No available meals."});
+    }
+    res.json(firstMeal);
```
```suggestion
    res.json(firstMeal[0]);
```
Even though there is only one row (meal) returned, the type of the variable is still an array, meaning you have to dereference the first value (`[0]`) to make sure you return an object, not an array.

kiko-bkr commented on api/src/index.js:
```diff
@@ -29,3 +29,70 @@ app.use("/api", apiRouter);
+    if(!lastMeal || lastMeal.length ===0) {
+      return res.status(404).json({ error: "No available meals."});
+    }
+    res.json(lastMeal);
```
Same here.

kiko-bkr reviewed (changes requested):
Hi! Your code is carefully written and clean! The only thing I have to highlight is that you should only be returning the actual resulting rows, whilst right now you just pass the result of `knex.raw()`, which also contains the schema. Once you fix that the code will be on point! ⚡

jianxinz233 committed:
update after review

jianxinz233 commented:
> Hi! Your code is carefully written and clean! The only thing I have to highlight is that you should only be returning the actual resulting rows, whilst right now you just pass the result of `knex.raw()`, which also contains the schema. Once you fix that the code will be on point! ⚡

Thank you for your review and all the suggestions, I kind of figured it out about the issues. And I just fixed and updated it 😊

kiko-bkr reviewed (approved):
Nicely done 👍
